scripts/pr/pr.py

```diff
@@ -260,16 +284,220 @@ def executeTask(self, args):
             task = self.tasks.get(taskName, Tasks.defaultScenario)
             task(self)
 
-            
+
+
+class HierarchicParserBuilder:
+    def __init__(self, routes):
+        self.parser = self.buildParser(routes)
+
+    def buildParser(self, routes):
+        p = argparse.ArgumentParser(description='Helper tool')
+
+        def helper(parser, structure):
+            func = structure.pop('func', None)
+            args = structure.pop('args', {})
+
+            for (name,params) in args.items():
+                parser.add_argument(name, **params)
+
+            if func:
+                parser.set_defaults(func=func)
+
+            subparser = None
+
+            for (name,subStruct) in structure.items():
+                subparser = orElse(subparser, parser.add_subparsers)
+
+                help = subStruct.pop('help', None)
+                parserForName = subparser.add_parser(name, help=help)
+                helper(parserForName, subStruct)
+
+            return subparser
+
+        helper(p, routes)
+        return p
+
+
+    def parse(self, args=None):
+        r = self.parser.parse_args(args)
+        return r.func(r)
+
+
+
+class Workflow:
+
+    def __init__(self, config, tasks):
+        self.config = config
+        self.tasks = tasks
+        self.commitMsg = {
+            'comment': self.commentCommit,
+            'transite': self.transiteCommit
```

Comment:
I might just do a global search-and-destroy for `transite` :)

Reply:
Oh, sure.